sdk/nodejs/appservice/kind.ts

```diff
@@ -0,0 +1,30 @@
+// Copyright 2016-2018, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+export module Kinds {
+    export const App:         Kind = "App";
+    export const Elastic:     Kind = "elastic";
+    export const FunctionApp: Kind = "FunctionApp";
+    export const Linux:       Kind = "Linux";
+    export const Windows:     Kind = "Windows";
+    export const Xenon:       Kind = "xenon";
```

Comment:
App Service Plans have several related properties and only some unique combinations of those are allowed. I expect people to get this wrong quite often, so maybe we could help with higher-level types? E.g. define something like

``` ts 
export namespace Plans {
    export const Consumption = {
        kind: "FunctionApp",
        sku: {
            tier: "Dynamic",
            size: "Y1",
        },
    };

    export const PremiumContainer2 = {
        kind: "xenon",
        isXenon: true,
        sku: {
            tier: "PremiumContainer",
            size: "PC2",
        },
    };
}
```

and then use them as

``` ts
const appServicePlan = new azure.appservice.Plan("asp", {
    resourceGroupName: resourceGroup.name,
    ...Plans.Consumption,
});
```

Reply:
I really like this idea - can you open a separate issue so we can talk about this and the design that you think would work? It would follow something like was requested here https://github.com/pulumi/pulumi-aws/issues/419